server/src/main/java/io/littlehorse/server/streams/taskqueue/OneTaskQueue.java

```diff
@@ -127,8 +127,8 @@ public boolean onTaskScheduled(ScheduledTaskModel scheduledTask) {
                 luckyClient = hungryClients.poll();
             } else {
                 // case 2
-                outOfCapacity = !pendingTasks.offer(scheduledTask);
-                return !outOfCapacity;
+                hasMoreTasksOnDisk = !pendingTasks.offer(scheduledTask) || hasMoreTasksOnDisk;
```

Comment:
This line was the problem. The other changes were just renaming the variable to make more sense.